src/Bundle/FrameworkBundle/src/Console/Application.js

```diff
@@ -0,0 +1,133 @@
+const BaseApplication = Jymfony.Component.Console.Application;
+const KernelInterface = Jymfony.Component.Kernel.KernelInterface;
+const Kernel = Jymfony.Component.Kernel.Kernel;
+const InputOption = Jymfony.Component.Console.Input.InputOption;
+
+/**
+ * @memberOf Jymfony.Bundle.FrameworkBundle.Console
+ */
+class Application extends BaseApplication {
+
+    /**
+     * Constructor.
+     *
+     * @param {Jymfony.Component.Kernel.KernelInterface} kernel A KernelInterface instance
+     */
+    __construct(kernel) {
+        /**
+         * @type {Jymfony.Component.Kernel.KernelInterface}
+         * @private
+         */
+        this._kernel = kernel;
+
+        /**
+         * @type {boolean}
+         * @private
+         */
+        this._commandsRegistered = false;
+
+        super.__construct('Jymfony', Kernel.VERSION);
+        this.definition.addOption(new InputOption('--env', '-e', InputOption.VALUE_REQUIRED, 'The environment name', kernel.environment));
+        this.definition.addOption(new InputOption('--no-debug', undefined, InputOption.VALUE_NONE, 'Switches off debug mode'));
+    }
+
+    /**
+     * Gets the Kernel associated with this Console.
+     *
+     * @return {Jymfony.Component.Kernel.KernelInterface} A KernelInterface instance
+     */
+    get kernel() {
+        return this._kernel;
+    }
+
+    /**
+     * Runs the current application.
+     *
+     * @param {Jymfony.Component.Console.InputInterface} input An Input instance
+     * @param {Jymfony.Component.Console.OutputInterface} output An Output instance
+     *
+     * @return {int} 0 if everything went fine, or an error code
+     *
+     * @protected
+     */
+    * _doRun(input, output) {
+        this._kernel.boot();
+        this.dispatcher = this._kernel.container.get('event_dispatcher');
+
+        return super._doRun(input, output);
```

Comment:
I'm no more convinced that this is the correct way to call a parent generator.
We should test it, but I think that a delegate yield is needed here.